src/sdk/module/storage.h

```diff
@@ -57,19 +62,30 @@ class StorageBucket {
 
   virtual void remove_all() = 0;
   virtual ~StorageBucket() = default;
+  virtual void destroy() {};
+
+  boost::asio::io_service::strand* get_owner_strand() { return owner_strand; };
+  void set_owner_strand(boost::asio::io_service::strand* strand) { owner_strand = strand; };
+
+ private:
+  boost::asio::io_service::strand* owner_strand = nullptr;
+
 };
 
 class Storage : public DestroyableRef<Storage> {
  public:
   /// create a bucket or find a existing bucket
-  virtual std::unique_ptr<StorageBucket> get_bucket(const string_& name) = 0;
+  virtual shared_ptr_<StorageBucket> get_shared_bucket(const string_& name) = 0;
+
+  shared_ptr_<StorageBucket> get_strand_bucket(const string_& name, boost::asio::io_service::strand* strand);
```

Comment:
let's use dsa strand instead of ioservice strand here, the input should be `LinkStrand&` and strand_bucket should get a ref from it
howerver, just pre declare it,  not include the real Strand class in this header.
the real class is only needed in the strand_bucket class

also, the return type of strand bucket should be a ref_ , not a shared_ptr